docs/msiv1_token_revocation.md

```diff
@@ -0,0 +1,86 @@
+# MSAL Support for MSI v1 Token Revocation and Capabilities
+
+---
+
+## Goals
+
+1. App developers and higher level SDKs like Azure SDK rely on the [CAE protocol](https://learn.microsoft.com/en-us/entra/identity-platform/app-resilience-continuous-access-evaluation?tabs=dotnet) for token revocation scenarios (`WithClaims`, `WithClientCapabilities`). 
+1. RPs are enabled to perform token revocation.
+1. A telemetry signal for eSTS exists to differentiate which apps are CAE enlightened.
+
+## Flow diagram - revocation event
+
+```mermaid
+sequenceDiagram
+    participant Resource
+    actor CX
+    participant MSAL
+    participant SF        
+    participant eSTS
+
+rect rgb(173, 216, 230)
+    CX->> Resource: 1. Call resource with "bad" token T
+    Resource->>CX: 2.HTTP 401 + claims C
+    CX->>CX: 3. Parse response, extract claims C
+    CX->>MSAL: 4. MSI.AcquireToken <br/> WithClaims(C) <br/> WithClientCapabilities("cp1")
+end
+rect rgb(215, 234, 132)
+    MSAL->>MSAL: 5. Get token T from cache. Assume it is the "bad" token.
+    MSAL->>SF: 6. Call MITS_endpoint?xms_cc=cp1&revoked_token=SHA256(T)
+    SF->>eSTS: 7. CCA.AcquireTokenForClient SN/I cert <br/> WithClientCapabilities(cp1) <br/> WithAccessTokenToRefresh(SHA256(T))
+end
+```
+
+Steps 1-4 fall to the Client (i.e. application using MSI directly or higher level SDK like Azure KeyVault). This is the **standard CAE flow**.
+Steps 5-7 are new and show how the RP propagates the revocation signal.
+
+
+> [!NOTE]  
+>  ClientCapabilities is an array of capabilities. In case the app developer sends multiple capabilities, these will be sent to the RP as `MITS_endpoint?xms_cc=cp1,cp2,cp3`. The RP MUST pass "cp1" (i.e. the CAE capabilitiy) if it is included.
+
+
+## Flow diagram - non-revocation event
+
+The client "enlightment" status is still propagated via the client capability "cp1".
+
+```mermaid
+sequenceDiagram
+    actor CX
+    participant MSAL
+    participant SF        
+    participant eSTS
+
+rect rgb(173, 216, 230)   
+    CX->>MSAL: 1. MSI.AcquireToken <br/> WithClientCapabilities("cp1")
+    MSAL->>MSAL: 2. Find and return token T in cache. <br/>If not found, goto next step.
+end
+rect rgb(215, 234, 132)    
+    MSAL->>SF: 3. Call MITS_endpoint?xms_cc=cp1
```

Comment:
Either simply remove `xms_cc`

```suggestion
    MSAL->>SF: 3. Call MITS_endpoint
```

or do it in the standard CAE way which is MSAL automatically converts xms_cc=cp1 into a `claims={"access_token": {"xms_cc": {"values": ["cp1"]}}}` and send that blob to the wire.

Reply:
MSAL needs to have that claims and capabilities separately, as they have different behavior. claims bypases cache, cp does not.

Reply:
Also, if memory serves, the call to MITS is a GET call. So the claims parameter could lead to an URL too long issue. Not sure if this is really a problem since it's ussually browsers that refuse long urls.